cmd/node-installer/install.go

```diff
@@ -36,9 +37,16 @@ var installCmd = &cobra.Command{
 	Run: func(_ *cobra.Command, _ []string) {
 		rootFs := afero.NewOsFs()
 		hostFs := afero.NewBasePathFs(rootFs, config.Host.RootPath)
-		restarter := containerd.NewRestarter()
 
-		if err := RunInstall(config, rootFs, hostFs, restarter); err != nil {
+		distro, err := DetectDistro(config, hostFs)
+		if err != nil {
+			slog.Error("failed to detect containerd config", "error", err)
+			os.Exit(1)
+		}
+
+		config.Runtime.ConfigPath = distro.ConfigPath
+
+		if err := RunInstall(config, rootFs, hostFs, distro.Restarter(preset.Env{ConfigPath: distro.ConfigPath, HostFs: hostFs})); err != nil {
```

Comment:
Not happy yet with how parts of the returned `distro` struct ends up in `config` and the restarter is passed as separate argument.